Library/Homebrew/rubocops/text.rb

```diff
@@ -62,6 +74,50 @@ def audit_formula(_node, _class_node, _parent_class_node, body_node)
           find_method_with_args(body_node, :system, "cargo", "build") do
             problem "use \"cargo\", \"install\", *std_cargo_args"
           end
+
+          find_every_method_call_by_name(body_node, :system).each do |m|
+            next unless parameters_passed?(m, /make && make/)
+
+            offending_node(m)
+            problem "Use separate `make` calls"
+          end
+
+          body_node.each_descendant(:dstr) do |dstr_node|
+            dstr_node.each_descendant(:begin) do |interpolation_node|
+              next unless interpolation_node.source.match?(/#\{\w+\s*\+\s*['"][^}]+\}/)
+
+              offending_node(interpolation_node)
+              problem "Do not concatenate paths in string interpolation"
+            end
+          end
+
+          find_strings(body_node).each do |n|
+            next unless regex_match_group(n, /JAVA_HOME/i)
+
+            next if @formula_name.match?(/^openjdk(@|$)/)
+
+            next if find_every_method_call_by_name(body_node, :depends_on).any? do |dependency|
+              dependency.each_descendant(:str).count.zero? ||
+              regex_match_group(dependency.each_descendant(:str).first, /^openjdk(@|$)/) ||
+              depends_on?(:java)
+            end
+
+            offending_node(n)
+            problem "Use `depends_on :java` to set JAVA_HOME"
+          end
+
+          find_strings(body_node).each do |n|
+            # Skip strings that don't start with one of the keywords
+            next unless regex_match_group(n, %r{^(bin|include|libexec|lib|sbin|share|Frameworks)/?})
```

Comment:
Could this do a direct method match?

Reply:
This is finding strings that start with any of the keywords, not methods

Reply:
Gotcha, my bad.